.github/workflows/ci.yml

```diff
@@ -66,7 +66,7 @@ jobs:
       if: >-
         github.event_name != 'schedule' &&
         matrix.os == 'ubuntu-latest'
-      uses: codecov/codecov-action@v3
+      uses: codecov/codecov-action@v4
```

Comment:
This needs to get revised to add a token given https://github.com/codecov/codecov-action/tree/v4.0.1?tab=readme-ov-file#usage

e.g.
```yaml
steps:
...
- uses: codecov/codecov-action@v4
  with:
    fail_ci_if_error: true # optional (default = false)
    files: ./coverage1.xml,./coverage2.xml # optional
    flags: unittests # optional
    name: codecov-umbrella # optional
    token: ${{ secrets.CODECOV_TOKEN }} # required
    verbose: true # optional (default = false)
```